artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -1516,6 +1541,9 @@ public synchronized RoutingStatus send(Transaction tx,
                                           final boolean direct,
                                           boolean noAutoCreateQueue,
                                           RoutingContext routingContext) throws Exception {
+      if (AuditLogger.isEnabled()) {
+         AuditLogger.sendMessage(this, getUsername(), tx, messageParameter, direct, noAutoCreateQueue, routingContext);
```

Comment:
Shouldn't this be done on PostOfficeImpl::send?

Then the same would apply to all the protocols.


Currently your implementation would ignore anything coming from Federation, MQTT AMQP or Stomp, since those will call PostOfficeImpl::send directly.

Reply:
If there is audit on this. I think there needs to be some audit level seperation. E.g. users may want to audit management actions but not want to audit log every message going through the broker

Reply:
I am -1 it for this reason to avoid this just being merged without this being addressed.

Reply:
Just to add i like the idea, just think there needs to be finer control on what to audit especially hot paths like send message and other bits.

Reply:
Please see:  org.apache.activemq.artemis.core.server.plugin.impl.LoggingActiveMQServerPlugin

It be good to extend this, rather than another implementation.

Reply:
> Shouldn't this be done on PostOfficeImpl::send?
> 
> Then the same would apply to all the protocols.
> 
> Currently your implementation would ignore anything coming from Federation, MQTT AMQP or Stomp, since those will call PostOfficeImpl::send directly.

The send() is auditing the message coming into the broker. So I think it can cover other protocols. right? (currently the audit doesn't record the messages going out but it logs the consumer creations).

Reply:
Hi @michaelandrepearce Just had a look at the plugin class my impression is it's a big too complicated for audit log. But let me think about it more. Also I agree the idea to fine control the logging on 'hot' paths so to give user a choice to turn off them particularly.
Thanks

Reply:
Whats complicated about it? You define plugin in broker xml and voila... 

The point being this is exactly the use case for plugins

Reply:
the log format is different. and the logging plugin serves a different purpose that auditing. (actually in amq5 they are totally different impls). and it brings in extra config.
for hot path (actually so far there is only one sending message) I can use log level to easily control it.
So I'd for the moment don't touch the logging plugin. 
wdyt?